pkg/access_token/access_token_file_test.go

```diff
@@ -19,138 +19,200 @@ func TestAccessTokenFile(t *testing.T) {
 	var tokenInFile, _ = NewAccessTokenFile(config)
 
 	Convey("Read", t, func() {
-		Convey("Returns true if Data output as expected", func() {
+
+		Convey("Given an access token with data", func() {
 			dataActual := []byte{'t', 'e', 's', 't'}
 			err := tokenInFile.Write(dataActual)
-			So(err, ShouldEqual, nil)
 
-			dataExpected, err := tokenInFile.Read()
-			eq := reflect.DeepEqual(dataActual, dataExpected)
+			Convey("Finishes without raising an error", func() {
+				So(err, ShouldEqual, nil)
+			})
+
+			Convey("When running Read method", func() {
+				dataExpected, err := tokenInFile.Read()
+
+				Convey("Finishes without raising an error", func() {
+					So(err, ShouldEqual, nil)
+				})
 
-			So(err, ShouldEqual, nil)
-			So(eq, ShouldEqual, true)
+				Convey("Returns the data that was written", func() {
+					eq := reflect.DeepEqual(dataActual, dataExpected)
+					So(eq, ShouldEqual, true)
+				})
+			})
 		})
 
-		Convey("Returns error if Data is nil", func() {
+		Convey("Given an access token's data is empty", func () {
 			tokenInFile.Data = nil
-			_, err := tokenInFile.Read()
 
-			So(err.Error(), ShouldEqual, "error reading access token, reason: data is empty")
+			Convey("When the token data is empty and running the Read method", func() {
+				_, err := tokenInFile.Read()
+
+				Convey("Raises an error that the data is empty", func() {
+					So(err.Error(), ShouldEqual, "error reading access token, reason: data is empty")
+				})
+			})
 		})
 	})
 
 	Convey("Write", t, func() {
-		Convey("Returns no error if Data input is not nil", func() {
+
+		Convey("Given an access token with data", func() {
 			dataActual := []byte{'t', 'e', 's', 't'}
-			tokenInFile.TokenFilePath = "/tmp/accessTokenFileWrite1"
-			err := tokenInFile.Write(dataActual)
 
-			So(err, ShouldEqual, nil)
-			// Check if file exits
-			_, err = os.Stat("/tmp/accessTokenFileWrite1")
-			So(err, ShouldEqual, nil)
+			Convey("When running the Write method, writes the access token to a file in the path defined", func() {
```

Comment:
i think this should be:
```
Convey("When running the Write method, ", func() {
	tokenInFile.TokenFilePath = "/tmp/accessTokenFileWrite1"
	err := tokenInFile.Write(dataActual)

	Convey("Finishes without raising an error", func() {
		So(err, ShouldEqual, nil)
	})

	Convey("Writes the access token to a file in the path defined", func() {
		_, err = os.Stat("/tmp/accessTokenFileWrite1")
		So(err, ShouldEqual, nil)
	})
})
```

Reply:
The last convey here doesn't do what you suggested the message should be.

`Writes the access token to a file in the path defined` should be when the actual writing is taking place. Here we are just checking if the file actually exists. What say you?

Reply:
maybe read the content of the file?

Reply:
Can't it be "Checks that the file exists"? If it fails we know that the file doesn't exist

Reply:
yes but then we don't check that the `write` operation succeeds

Reply:
I do see value in this test
Added the following:
```
Convey("Checks that the file exists in the path defined", func (){
	// Check if file exists
	_, err = os.Stat("/tmp/accessTokenFileWrite1")
	So(err, ShouldEqual, nil)
})

Convey("When running Read method", func() {
	dataExpected, _ := tokenInFile.Read()

	// Confirm data was written
	Convey("Returns the data the was written to the file", func() {
		eq := reflect.DeepEqual(dataActual, dataExpected)
		So(eq, ShouldEqual, true)
	})
})
```

Reply:
i added in previous commit the read method that confirms that the data was written to the file